src/main/java/org/opensearch/knn/quantization/util/BitPackingUtils.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.knn.quantization.util;
+
+import lombok.experimental.UtilityClass;
+
+import java.util.List;
+
+/**
+ * Utility class for bit packing operations.
+ * Provides methods for packing arrays of bits into byte arrays for efficient storage or transmission.
+ */
+@UtilityClass
+public class BitPackingUtils {
```

Comment:
nit: A suffix `Utils` generally opens up a possibility of misuse. Lets try to define the name in a way that it constraints the responsibility. Consider `BitPacker` whole responsibility can be pack and unpack as its an inverse function